app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -37,22 +50,29 @@ def state
 
     def start
       sdk.start_vm(@azure_vm.resource_group, name)
+      true
     end
 
     def stop
       sdk.stop_vm(@azure_vm.resource_group, name)
+      true
     end
 
     def vm_status
       sdk.check_vm_status(@azure_vm.resource_group, name)
     end
 
     def network_interface_card_ids
+      return nil unless @azure_vm.network_profile
       nics = @azure_vm.network_profile.network_interfaces
       nics.map(&:id)
     end
 
     def provisioning_ip_address
+      public_ip_address || private_ip_address
+    end
+
+    def public_ip_address
       interfaces.each do |nic|
         nic.ip_configurations.each do |configuration|
           next unless configuration.primary
```

Comment:
I'm returning nil here if the public ip is not present: [line 86](https://github.com/theforeman/foreman_azure_rm/pull/43/files/#diff-73772f98407930e1c343b4b76ca2963aR86)

Reply:
What I meant to say was that instead of having `if.. else` loop why not to mention something like -

`return nil if configuration.public_ipaddress.blank?
`

Reply:
done.